packages/backend/src/middlewares/validator.ts

```diff
@@ -1,16 +1,25 @@
 import { Request } from 'express'
 import { AnyZodObject, z, ZodError } from 'zod'
-import { BadRequestException } from '../errors/BadRequestException'
+import { BadRequestException } from '../shared/models/errors/BadRequestException'
 
 export async function zParse<T extends AnyZodObject>(
   schema: T,
   req: Request
 ): Promise<z.infer<T>> {
   try {
-    return schema.parseAsync(req.body)
+    const res = await schema.parseAsync(req.body)
+    return res
   } catch (error) {
     if (error instanceof ZodError) {
-      throw new BadRequestException(error.errors[0].message)
+      const errors = error?.errors?.reduce((acc, error) => {
+        const newAcc = {
+          ...acc,
+          [error.path[0]]: error.message
+        }
+        return newAcc
+      }, {})
+
+      throw new BadRequestException('Invalid input', errors)
```

Comment:
You can use `flatten` for this (https://github.com/colinhacks/zod/blob/master/ERROR_HANDLING.md#flattening-errors)

Reply:
I used flatten, and also tried giving it a custom mapper as in these docs:
![image](https://user-images.githubusercontent.com/111863110/221944001-dbbd9353-ab2c-415e-9a7d-b942a01033fc.png)

However, it did not help much, since I was forced to return an object for each fieldError key, instead of the string that we wanted. 

My latest code maps the arrays to strings.